### PR TITLE
[#80] Set netty and akka log level from service level 

### LINKED ISSUE #80: Set netty and akka log level from service level
See io.github.ztkmkoo.dss.core.network.rest.DssRestChannel

``` java
public class DssRestChannel {
    public DssRestChannel() {
    }

    public Channel bind(ServerBootstrap serverBootstrap, DssRestChannelProperty dssRestChannelProperty, DssRestChannelInitializer dssRestChannelInitializer) throws InterruptedException {
        Objects.requireNonNull(serverBootstrap);
        Objects.requireNonNull(dssRestChannelProperty);
        Objects.requireNonNull(dssRestChannelInitializer);
        return ((ServerBootstrap)((ServerBootstrap)serverBootstrap.channel(NioServerSocketChannel.class)).handler(new LoggingHandler(LogLevel.DEBUG))).childHandler(dssRestChannelInitializer).bind(dssRestChannelProperty.getHost(), dssRestChannelProperty.getPort()).sync().channel();
    }
}
```

It config netty loglevel default as debug level.
By the way, i can config the log level in the service project with create new logback.xml file(so we can see thenetty debug log), but it is necessary to make config log level possible.

How bout implementing a new log level enumeration mapping netty log level(io.netty.handler.logging.LogLevel) and let's pass the parameter to DssRestServer constructure.

### PR DESCRIPTION
**Pull request by vincentj2**
resolves #80

-  Add DssLogLevel enum
-  mapping netty LogLevel to DssLogLevel

### CODE REVIEW

File: dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java [hunk 1]
```diff
@@ -17,10 +17,11 @@
...
     private final String host;
     private final int port;
-
+    private final DssLogLevel dssLogLevel;
     @Builder
-    private DssRestChannelProperty(String host, int port) {
+    private DssRestChannelProperty(String host, int port, DssLogLevel dssLogLevel) {
         this.host = host;
         this.port = port;
+        this.dssLogLevel = dssLogLevel;
```
Comment (ztkmkoo) [inline dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:25]: @vincentj2 
Hi~

It looks you do not set default value of dssLogLevel and cause ci failed

Comment (vincentj2) [reply dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:25]: hi

yes.. that's my mistake.   I'll fit it!

Comment (vincentj2) [reply dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:25]: I fix line 25 
` this.dssLogLevel = dssLogLevel; `
to
` this.dssLogLevel = DssLogLevel.DEBUG; `

Is it okey?

Comment (ztkmkoo) [reply dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:25]: Nope.
Following better, i think.
``` java
this.dssLogLevel = Objects.nonNull(dssLogLevel) ? dssLogLevel : DssLogLevel.DEBUG;
```

Comment (vincentj2) [reply dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:25]: I think that will be better!!
thank you :)

**Comment by ztkmkoo** [comment]
@vincentj2 
plz update your branch to latest

**Comment by vincentj2** [comment]
@ztkmkoo 
I'm done

**Comment by ztkmkoo** [comment]
> 
> 
> @ztkmkoo
> I'm done

Greate. i will merge this pr after the review.

File: dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java [hunk 2]
```diff
@@ -17,10 +18,11 @@
     private static final long serialVersionUID = -6904414304979344025L;
     private final String host;
     private final int port;
-
+    private final DssLogLevel dssLogLevel;
     @Builder
```
Comment (Alencion) [inline dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:22]: Hi, @vincentj2 

please add new line between fields and methods.

File: dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java [hunk 3]
```diff
@@ -17,10 +18,11 @@
...
-    private DssRestChannelProperty(String host, int port) {
+    private DssRestChannelProperty(String host, int port, DssLogLevel dssLogLevel) {
         this.host = host;
         this.port = port;
+        this.dssLogLevel = Objects.nonNull(dssLogLevel) ? dssLogLevel : DssLogLevel.DEBUG;;
```
Comment (Alencion) [inline dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:26]: I don't know well. 

``` java
DssRestChannelProperty
                .builder()
                .host(host)
                .port(port)
                .build();
```
Like this. If dssLogLevel is not set, can it be set to dssLogLevel.DEBUG? 

And I think, it seems to need that how to set DssLogLevel.
``` java
// in DssRestServer
public DssRestServer(String host, int port,) {
    this(host, port, DssLogLevel.DEBUG ,false, null);
} 

public DssRestServer(String host, int port, DssLogLevel dssLogLevel) {
    this(host, port, dssLogLevel, false, null);
}
```
and please delete semicolon.

File: dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java [hunk 4]
```diff
@@ -17,10 +18,12 @@
...
     private final String host;
     private final int port;
+    private final DssLogLevel dssLogLevel;
 
     @Builder
-    private DssRestChannelProperty(String host, int port) {
+    private DssRestChannelProperty(String host, int port, DssLogLevel dssLogLevel) {
         this.host = host;
         this.port = port;
+        this.dssLogLevel = Objects.nonNull(dssLogLevel) ? dssLogLevel : DssLogLevel.DEBUG;
```
[the remark this answers is not in the document]
Comment (RulLu16) [reply dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:27]: I agree with @ztkmkoo.

Just using that code(this.dssLogLevel = dssLogLevel;) is not a problem now. But as we add more code in this project, we can use this method with null and can cause error. So I think preventing the possibility of error is better than fixing the error later.

**Review by RulLu16 — APPROVED** [review]

File: dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Alencion) [reply dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:27]: Oh. I understand. 

Thank you for your kind explanation. @ztkmkoo @RulLu16

[the remark this answers is not in the document]
Comment (ztkmkoo) [reply dss-core/src/main/java/io/github/ztkmkoo/dss/core/network/rest/property/DssRestChannelProperty.java:27]: @Alencion @RulLu16 
Thank you two!

**Pull request merged**